cmd/xl-sets.go

```diff
@@ -522,11 +523,13 @@ func (s *xlSets) MakeBucketWithLocation(ctx context.Context, bucket, location st
 	}
 
 	errs := g.Wait()
-	// Upon any error we try to undo the make bucket operation if possible
-	// on all sets where it succeeded.
+	// Upon any error, except bucket already exists, we try to undo
+	// the make bucket operation if possible on all sets where it succeeded.
 	for _, err := range errs {
 		if err != nil {
-			undoMakeBucketSets(bucket, s.sets, errs)
+			if !errors.As(err, &BucketExists{}) {
+				undoMakeBucketSets(bucket, s.sets, errs)
```

Comment:
Return error here.

Continue otherwise.

Reply:
@klauspost actually, I think what we need to do is to reduce the error of all MakeBucket calls on all sets, then decide based on the reduced error if we should revert the bucket creation or not.

For example:
* If at least len(sets)/2+1 of calls return nil, then it is success and do nothing 
* If len(sets)/2+1 return BucketExists, then returns also BucketExists
* We should probably try to undo the bucket creation otherwise

Reply:
> * We should probably try to undo the bucket creation otherwise

This won't work actually.. The issue here is that, in this specific case that we are trying to fix, MakeBucket is called in server's startup. MakeBucket() will most likely return an error due to not yet met write quorumsome nodes are still not up.